### PR TITLE
transmission: Export new WireEvent type

### PR DESCRIPTION
**Pull request by dcarley**
Which represents the event that Honeycomb's API accepts. I'd like to use
this in a mock/load testing server that takes Honeycomb batched events.
Having it exported would mean that I don't need to redefine it myself or
miss breaking changes. It could also be used by samproxy:

- https://github.com/honeycombio/samproxy/blob/3418d37765914ce531df96ccef0ecf3d0a5421e7/route/route.go#L494-L498

### CODE REVIEW

File: transmission/event.go [hunk 1]
```diff
@@ -11,6 +11,12 @@ import (
 	"github.com/vmihailenco/msgpack/v4"
 )
 
+type WireEvent struct {
+	Data       MarshallableMap `json:"data" msgpack:"data"`
+	SampleRate uint            `json:"samplerate,omitempty" msgpack:"samplerate,omitempty"`
+	Timestamp  *time.Time      `json:"time,omitempty" msgpack:"time,omitempty"`
```
Comment (dcarley) [inline transmission/event.go:17]: I kept the original order here, for compatibility with the tests, but ideally I'd expected to define and render in the opposite order:
- Timestamp
- SampleRate
- Data

**Comment by maplebed** [comment]
Can you say more about what benefits this gives that are different from using an alternative implementation of the `Sender` interface within libhoney? 

My interpretation is that the goal here is to use the library when writing code that _receives_ Honeycomb events rather than _sending_ events, is that right? My first reaction is to want to think a bit more about what that looks like (a library for receiving events) rather than only adding an exported type. It's an intriguing idea though, what a receiver might look like or what functionality it might provide.

**Comment by dcarley** [comment]
> Can you say more about what benefits this gives that are different from using an alternative implementation of the `Sender` interface within libhoney?

Don't think this is applicable after the next question, but correct me if I've misunderstood.

> My interpretation is that the goal here is to use the library when writing code that _receives_ Honeycomb events rather than _sending_ events, is that right?

Yeah, that's right. I was originally using `transmission.Event` until I realised that the `time`/`timestamp` fields have different names. So I had to dig further into the `transmission` package and the [description in the API docs](https://docs.honeycomb.io/api/events/#batched-events) to get it right.

> My first reaction is to want to think a bit more about what that looks like (a library for receiving events) rather than only adding an exported type. It's an intriguing idea though, what a receiver might look like or what functionality it might provide.

It seems relatively niche. Are you aware of any other use cases that have come up previously?

Basically I just want some cheap type safety 😄

**Comment by MikeGoldsmith** [comment]
We’re going to close this for now. Feel free to re-open, if it’s still relevant.
